kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java

```
@@ -582,4 +595,50 @@ public Optional<ApplicationSection> section() {
+                        .filter(Statement::isExpressionStmt)
+                        .flatMap(stmt -> stmt.getChildNodes().stream())
+                        .filter(AssignExpr.class::isInstance)
+                        .map(AssignExpr.class::cast)
+                        .filter(assignExpr -> assignExpr.getTarget() instanceof FieldAccessExpr field &&
+                                BUSINESS_CALENDAR_FIELD_NAME.equals(field.getNameAsString()))
+                        .forEach(assignExpr -> assignExpr.setValue(expression))));
+
+        storeFile(PRODUCER_TYPE, generator.generatedFilePath(), compilationUnit.toString());
+    }
+
+    private Optional<Expression> deriveBusinessCalendarDeclarationStatement(String businessCalendarClassName) {
+        Class<?> businessCalendarClass;
+        boolean validBusinessCalendar = false;
+        ObjectCreationExpr declarationStmt = null;
+        if (Objects.isNull(businessCalendarClassName)) {
+            return Optional.empty();
+        }
+        try {
+            businessCalendarClass = Class.forName(businessCalendarClassName, true, context().getClassLoader());
+        } catch (ClassNotFoundException e) {
+            LOGGER.error("Custom Business Calendar class {} not found.\nGenerating default business calendar", businessCalendarClassName);
+            return Optional.empty();
+        }
+        if (!businessCalendarClass.isInterface()) {
+            validBusinessCalendar = BusinessCalendar.class.isAssignableFrom(businessCalendarClass);
+        }
+        if (!validBusinessCalendar) {
+            LOGGER.error("Custom Business Calendar class {} does not implement BusinessCalendar interface.\nGenerating default business calendar", businessCalendarClassName);
+        } else {
+            declarationStmt = new ObjectCreationExpr(null, StaticJavaParser.parseClassOrInterfaceType(businessCalendarClassName), NodeList.nodeList());
+        }
+        return Optional.ofNullable(declarationStmt);
```

Please read carefully the logic you implemented.
This `deriveBusinessCalendarDeclarationStatement(String)` method is invoked only when `businessCalendarClassName` is different then `null`, so it should never return an empty object: in case, it should throw an exception and fail the build, as explained in previous comment